src/class/ResponseHeader/ResponseHeader.cpp

```diff
@@ -0,0 +1,224 @@
+
+#include <ResponseHeader/ResponseHeader.hpp>
+
+/*
+** ------------------------------- CONSTRUCTOR --------------------------------
+*/
+
+ResponseHeader::ResponseHeader(int code, const ErrorPage &default_page)
+{
+	defaultConstructor();
+	setStatusCode(code);
+	if (code >= 400)
+	{
+		setContentType(".html");
+		setConnection("close");
+		defaultErrorPage(code, default_page);
+		if (code == 405)
+			_header[CONNECTION].name = "Allow: ";
+	}
+}
+
+ResponseHeader::ResponseHeader(const ResponseHeader &src)
+	: _header(src.getHeader())
+{
+}
+
+const char *ResponseHeader::InvalidDefaultPage::what() const throw()
+{
+	return "Invalid default page";
+}
+
+/*
+** -------------------------------- DESTRUCTOR --------------------------------
+*/
+
+ResponseHeader::~ResponseHeader()
+{
+}
+
+/*
+** --------------------------------- OVERLOAD ---------------------------------
+*/
+
+ResponseHeader &ResponseHeader::operator=(ResponseHeader const &rhs)
+{
+	if (this != &rhs)
+	{
+		this->_header = rhs.getHeader();
+	}
+	return *this;
+}
+
+std::ostream &operator<<(std::ostream &o, ResponseHeader const &i)
+{
+	o << i.getResponse();
+	return o;
+}
+
+/*
+** --------------------------------- METHODS ----------------------------------
+*/
+
+void ResponseHeader::defaultConstructor()
+{
+
+	for (field f = STANDARD; f <= BODY; f = static_cast<field>(static_cast<int>(f) + 1))
+	{
+		HeaderFields defaultHeader;
+		_header.insert(std::make_pair(f, defaultHeader));
+	}
+	_header[STANDARD].name = "HTTP/1.1 ";
+	_header[STANDARD].value = "";
+	_header[SERVER].name = "Server: ";
+	_header[SERVER].value = "webserv";
+	_header[DATE].name = "Date: ";
+	_header[DATE].value = getCurrentDate();
+	_header[CONNECTION_TYPE].name = "Content-Type: ";
+	_header[CONNECTION_TYPE].value = "";
+	_header[CONTENT_LENGHT].name = "Content-Length: ";
+	_header[CONTENT_LENGHT].value = "";
+	_header[CONNECTION].name = "Connection: ";
+	_header[BODY].name = "\r\n";
+	_header[BODY].value = "";
+	setContentType(".txt");
+	setContentLength(0);
+	setConnection("keep-alive");
+}
+
+void ResponseHeader::defaultErrorPage(int code, const ErrorPage &default_page)
+{
+	HTTPStatusCode default_code = default_page.getStatus();
+	if (code == default_code.get())
+	{
+		std::string path = default_page.getPath().get();
+		std::ifstream file(path.c_str());
```

Comment:
another idea haha
maybe when the app start we can load all of the ErrorPages to the mem so you will not need to read a files at this point

something like:
```
error_page = error_page_list.find(code);
setBody(error_page.getLoadedPage());
```

Reply:
I like the idea. That will make our server faster while it is running. The only thing is that it would no longer behave the same as nginx.

Reply:
so nginx is reading the file every time?
if a modify it while it is runing it will change?

Reply:
yes, it does